api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationOAuth.java

```diff
@@ -105,6 +106,17 @@ public void setCheckIssuer(boolean checkIssuer) {
         this.checkIssuer = checkIssuer;
     }
 
+    @Description("Enable or disable audience checking. If audience checking is enabled the `clientId` also has to be configured. " +
```

Comment:
So, this checks that the `aud == clientID`? If so, should we make it clear here in the description? For me this is a bit cryptic right now ... I know audience is in the token. But it is not clear what is it checked for. 

I noticed this is explained in one of the other sections. Bu might be worth adding it here as well?

Reply:
> If audience checking is enabled the `clientId` also has to be configured.

Configured where? The documentation should be explicit.

Reply:
_Enable or disable audience checking. Audience checks identify the recipients of tokens. If audience checking is enabled, the Kafka broker will reject tokens that do not have its `clientId` in their `aud` (audience) claim. Default value is `false`._

Reply:
> > If audience checking is enabled the `clientId` also has to be configured.
> 
> Configured where? The documentation should be explicit.

It's a config attribute of 'oauth' authentication called `clientId` at the same level as `validIssuerId`, `userNameClaim`, and `checkAudience`. Not sure how to make it more clear since the generated docs have these in the same set of config attributes, alphabetically sorted. When you read this description in the generated KafkaCR documentation under `checkAudience` attribute, you can just scroll a bit further to read about `clientId` config attribute. Is there a way to create links from the description to the attribute entry in the docs?